### DISCUSSION TITLE
Addon API to configure the sidebar

### DISCUSSION THREAD

**Discussion by jwir3** [storybookjs/storybook#23096] [Ideas]
First, I apologize if this is not a new feature request. I did look through the existing issues, but I was unable to find anything that clearly allowed for what I wanted to do.

**Is your feature request related to a problem? Please describe**
I have multiple versions of design systems that are evolving over time, and I want a way to clearly communicate to the user which version of the design system they are currently viewing. My thinking is that this would be ideally situated underneath the logo on the left sidebar of the storybook page.

**Describe the solution you'd like**
I'd like to be able to write an addon that adds components to the sidebar, including the header portion at the top of the sidebar. Perhaps more descriptive would be the ability to somehow hook into the sidebar as a component, with the children of the sidebar being the React components that are included, which would likely include not just the `<Header>` and top-level components that are children of the `<Sidebar>`, but also the objects that are mapped when using `renderLabel`.

**Describe alternatives you've considered**
I've tried to hack `renderLabel()` to do what I want, but it doesn't actually iterate over the non-component menu items (e.g. the `<Header>`, `<SidebarLogo>`, etc... I've also tried writing an addon, but, as is the topic of this issue, there doesn't seem to be a way to access the Sidebar, only panels, tabs, and tools on the toolbar.

**Are you able to assist to bring the feature to reality?**
Yes, I can do this, as I'm fluent in Javascript and React, but I'd need some guidance on how to start working on the storybook codebase. (I'm just a user at this point).

**Additional context**
None.

**Comment by shilman** [comment]
Related #7711
